src/Facebook/FileUpload/FacebookResumableUploader.php

```diff
@@ -0,0 +1,208 @@
+ * Policies [http://developers.facebook.com/policy/]. This copyright notice
+ * shall be included in all copies or substantial portions of the software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+namespace Facebook\FileUpload;
+
+use Facebook\Exceptions\FacebookResponseException;
+use Facebook\Exceptions\FacebookSDKException;
+use Facebook\Exceptions\FacebookUploadException;
+use Facebook\Exceptions\FacebookUploadTransferException;
+use Facebook\FacebookApp;
+use Facebook\FacebookClient;
+use Facebook\FacebookRequest;
+
+/**
+ * Class FacebookResumableUploader
+ *
+ * @package Facebook
+ */
+class FacebookResumableUploader
+{
+    /**
+     * @var FacebookApp
+     */
+    protected $app;
+
+    /**
+     * @var string
+     */
+    protected $accessToken;
+
+    /**
+     * @var FacebookClient The Facebook client service.
+     */
+    protected $client;
+
+    /**
+     * @var int Max retry times
+     */
+    protected $maxTransferTries;
+
+    /**
+     * @param FacebookApp $app
+     * @param string $accessToken
+     * @param int $maxTransferTries
+     */
+    public function __construct(FacebookApp $app, $accessToken, $maxTransferTries = 5)
+    {
+        $this->accessToken = $accessToken;
+        $this->app = $app;
+        $this->client = new FacebookClient();
+        $this->maxTransferTries = $maxTransferTries;
+    }
+
+    /**
+     * Upload - phase start
+     *
+     * @param $endpoint
+     * @param $filePath
+     * @return FacebookTransferChunk
+     *
+     * @throws FacebookUploadException
+     * @throws FacebookSDKException
+     */
+    public function start($endpoint, $filePath)
+    {
+        $fileSize = filesize($filePath);
+
+        $startReqParams = [
+            'upload_phase' => 'start',
+            'file_size' => $fileSize,
+        ];
+
+        $request = new FacebookRequest(
+            $this->app,
+            $this->accessToken,
+            'POST',
+            $endpoint,
+            $startReqParams
+        );
+
+        try {
+            $res= $this->client->sendRequest($request)->getDecodedBody();
+
+            $firstTransferChunk = new FacebookTransferChunk(
+                $filePath,
+                $res['upload_session_id'],
+                $res['video_id'],
+                $res['start_offset'],
+                $res['end_offset'] - $res['start_offset']
+            );
+
+            return $firstTransferChunk;
+        } catch (FacebookResponseException $e) {
+            throw new FacebookUploadException($e->getResponse());
```

Comment:
I'd just add the resumable error codes to the [error handler](https://github.com/facebook/facebook-php-sdk-v4/blob/5.0-dev/src/Facebook/Exceptions/FacebookResponseException.php#L69-L130) and have it throw a `FacebookResumableUploadException` or something. :)